libraries/AP_Mount/AP_Mount_Gremsy.cpp

```diff
@@ -0,0 +1,318 @@
+                ((_instance == 1) && (compid == MAV_COMP_ID_GIMBAL2))) {
+                _found_gimbal = true;
+                _sysid = sysid;
+                _compid = compid;
+                _chan = chan;
+            }
+        } else {
+            // have not yet found a gimbal so return
+            return;
+        }
+    }
+
+    // request GIMBAL_DEVICE_INFORMATION
+    if (!_got_device_info) {
+        uint32_t now_ms = AP_HAL::millis();
+        if (now_ms - _last_devinfo_req_ms > 1000) {
+            _last_devinfo_req_ms = now_ms;
+            request_gimbal_device_information(_chan, _sysid, _compid);
+        }
+        return;
+    }
+
+    // start sending autopilot attitude to gimbal
+    if (start_sending_attitude_to_gimbal()) {
+        _initialised = true;
+    }
+}
+
+// handle GIMBAL_DEVICE_INFORMATION message
+void AP_Mount_Gremsy::handle_gimbal_device_information(const mavlink_message_t &msg)
+{
+    // exit immediately if this is not our message
+    if (msg.sysid != _sysid || msg.compid != _compid) {
+        return;
+    }
+
+    mavlink_gimbal_device_information_t info;
+    mavlink_msg_gimbal_device_information_decode(&msg, &info);
+
+    // set parameter defaults from gimbal information
+    _state._roll_angle_min.set_default(degrees(info.roll_min) * 100);
+    _state._roll_angle_max.set_default(degrees(info.roll_max) * 100);
+    _state._tilt_angle_min.set_default(degrees(info.pitch_min) * 100);
+    _state._tilt_angle_max.set_default(degrees(info.pitch_max) * 100);
+    _state._pan_angle_min.set_default(degrees(info.yaw_min) * 100);
+    _state._pan_angle_max.set_default(degrees(info.yaw_max) * 100);
+
+    const uint8_t fw_ver_major = info.firmware_version & 0x000000FF;
+    const uint8_t fw_ver_minor = (info.firmware_version & 0x0000FF00) >> 8;
+    const uint8_t fw_ver_revision = (info.firmware_version & 0x00FF0000) >> 16;
+    const uint8_t fw_ver_build = (info.firmware_version & 0xFF000000) >> 24;
+
+    // display gimbal info to user
+    gcs().send_text(MAV_SEVERITY_CRITICAL, "Mount: %s %s fw:%u.%u.%u.%u",
+            info.vendor_name,
+            info.model_name,
+            (unsigned)fw_ver_major,
+            (unsigned)fw_ver_minor,
+            (unsigned)fw_ver_revision,
+            (unsigned)fw_ver_build);
+
+    _got_device_info = true;
+}
+
+// handle GIMBAL_DEVICE_ATTITUDE_STATUS message
+void AP_Mount_Gremsy::handle_gimbal_device_attitude_status(const mavlink_message_t &msg)
+{
+    // exit immediately if this is not our message
+    if (msg.sysid != _sysid || msg.compid != _compid) {
+        return;
+    }
+
+    // take copy of message so it can be forwarded onto GCS later
+    mavlink_msg_gimbal_device_attitude_status_decode(&msg, &_gimbal_device_attitude_status);
+}
+
+// request GIMBAL_DEVICE_INFORMATION message
+void AP_Mount_Gremsy::request_gimbal_device_information(mavlink_channel_t chan, uint8_t dev_sysid, uint8_t dev_compid)
```

Comment:
there is, and should be, no need for the dev_sysid and dev_compid, the gimbal's ids should be known by now (and sysid should be the autopilot's sysid anyway)